wrangler-api/src/main/java/io/cdap/wrangler/api/Executor.java

```diff
@@ -80,5 +82,18 @@ O execute(I rows, ExecutorContext context)
    * correct at this phase of invocation.
    */
   void destroy();
-}
 
+  /**
+   * This method is used to get the updated schema of the data after the directive's transformation has been applied.
+   * @implNote By default, returns a null and the schema is inferred from the data when necessary.
+   * <p>For consistent handling, override for directives that perform column renames,
+   * column data type changes or column additions with specific schemas.</p>
+   * @param inputSchema input {@link Schema} of the data before transformation
+   * @return output {@link Schema} of the transformed data
+   */
+  @Nullable
+  default Schema getOutputSchema(Schema inputSchema) {
```

Comment:
What's the plan for directives that generate output schema based on input data (e.g. string parsing into multiple fields)? Are we going to rely on the detection in this case?

Reply:
Yes, the detection will take care of that. Even if we add logic to the directive, it will be redundant as it will be the same schema generation that will be done

Reply:
Reopening, please see previous comment